[studio-ui] Edit Panel Fixes & Parsing Repeats content type fields

aachavarria opened the pull request:
### Ticket reference or full description of what's in the PR

aachavarria committed:
Fixing issues related to drag components

aachavarria committed:
adding Disassociate vs Delete

aachavarria committed:
updating modelHierarchyMap after sort

aachavarria committed:
Updating element index on sort

aachavarria committed:
Updating updateModel logic

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into 4.0.0/stability

 Conflicts:
	ui/guest/src/classes/ContentController.ts

aachavarria committed:
Fixing undefined value when registers components

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into 4.0.0/stability

aachavarria committed:
Adding hierarchyMap to preview.guest store
Updating findParentModelId
Adding ParsingFields recursive function

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into 4.0.0/stability

rart commented on ui/app/src/components/PreviewAddressBar/PreviewAddressBar.tsx:
```diff
@@ -160,9 +160,11 @@ export function PreviewAddressBar(props: AddressBarProps) {
+          <IconButton onClick={onRefresh} size="large" disabled={disabled}>
+            <RefreshRounded />
+          </IconButton>
+        </div>
```
```suggestion
        <span>
          <IconButton onClick={onRefresh} size="large" disabled={disabled}>
            <RefreshRounded />
          </IconButton>
        </span>
```

rart commented on ui/app/src/utils/model.ts:
```diff
@@ -14,10 +14,13 @@
 import LookupTable from '../models/LookupTable';
+import { ModelHierarchyMap } from './content';
+import { forEach } from './array';
+import * as Model from '@craftercms/studio-guest/build_tsc/utils/model';
```
use local path based import

aachavarria replied:
I don't get it

aachavarria replied:
pending

aachavarria committed:
Fixing sort items inside repeat groups with no simple index

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into 4.0.0/stability

 Conflicts:
	ui/app/src/components/PreviewAddressBar/PreviewAddressBar.tsx

aachavarria committed:
Fixing issues on items ids inside of repeat group with no simple fieldid

aachavarria committed:
WIP dlete

rart commented on ui/guest/src/store/reducers/root.ts:
```diff
@@ -129,6 +129,7 @@ const reducer = createReducer(initialState, {
         return { ...state, highlighted: { [record.id]: highlight } };
       } else if (highlightMode === HighlightMode.MOVE_TARGETS) {
         const iceId = record.iceIds[0];
+
```
```suggestion
```

rart commented on ui/guest/src/classes/ICERegistry.ts:
```diff
@@ -356,20 +359,30 @@ export function isMovableType(id: number): boolean {
+    if (isSimple(fieldId)) {
+      // Can be...
+      // - Field of a component (possible move target)
+      // - Field of a repeat (certain move target)
```
```suggestion
```

rart commented on ui/guest/src/react/GuestProxy.tsx:
```diff
@@ -102,18 +102,45 @@ export function GuestProxy() {
       // proceeding to registering again. Registering immediately after de-registering on the same
       // loop causes misjudging ICE records refCounts due to displaced collection items.
+
+      function updateRepeatGroupsItems(element, fieldId, index, newIndex, array) {
```
`array` needs a better name

rart commented on ui/guest/src/classes/ICERegistry.ts:
```diff
@@ -78,6 +78,9 @@ let rid = 0;
 
+// @ts-ignore
+window.registry = registry;
+
```
x

rart commented on ui/guest/src/classes/ContentController.ts:
```diff
@@ -690,12 +721,12 @@ export function deleteItem(modelId: string, fieldId: string, index: number | str
+  //   fieldId,
+  //   index,
+  //   parentModelId: getParentModelId(modelId, models, modelHierarchyMap)
+  // });
```
👆

rart reviewed (commented):
Let's review together.

aachavarria committed:
Fixing PR Comments

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into 4.0.0/stability

aachavarria committed:
Creating getIndexMetaData to abstract functionality
Creating updateHierarchyMapIndexes to abstract functionality
Fixing setCollection

aachavarria committed:
Fixing issues when deleting items on repeat groups/ node selectors

aachavarria committed:
Fixes on insert Components / instances

aachavarria committed:
Creting and Adding new inserted item to modelHierarchyMap

rart commented on ui/guest/src/classes/ContentController.ts:
```diff
@@ -671,11 +717,32 @@ export function deleteItem(modelId: string, fieldId: string, index: number | str
+    updateHierarchyMapIndexes(result);
   } else {
-    // TODO: check modelHierarchyMap[item].children
+    const modelsToDeleted = [];
```
```suggestion
    const modelsToDelete = [];
```

rart commented:
@aachavarria please pull and fix conflicts

aachavarria committed:
typo

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into 4.0.0/stability

 Conflicts:
	ui/app/src/utils/model.ts
	ui/guest/src/utils/ice.ts

aachavarria committed:
new functions updateHierarchyMapIndexesFromCollection and updateHierarchyMapIndexesFromPosition

aachavarria committed:
Fixes issues when moving items

aachavarria committed:
Fixing repeat item dropzone

aachavarria committed:
Rollback index files

aachavarria committed:
Fixes related to contentType required fields
